src/litserve/server.py

```diff
@@ -472,3 +472,95 @@ def setup_auth(self):
         if LIT_SERVER_API_KEY:
             return api_key_auth
         return no_auth
+
+
+@asynccontextmanager
+async def multi_server_lifespan(app: FastAPI, servers: List[LitServer]):
+    """Context manager to handle the lifespan events of multiple FastAPI servers."""
+    # Start lifespan events for each server
+    app.response_queue_id = 0
+    async with AsyncExitStack() as stack:
```

Comment:
I couldn't find a reason why it is looking for `response_queue_id` in the main app. This issue seems to occur specifically in Linux environments and was initially the cause of test failures.

<details>
<summary>Reproduce</summary>

```python
# Open in codespace
# Hide or remove line 481;     app.response_queue_id = 0
pip install .
python tests/multiple_litserver.py
```
</details>

```sh
ERROR:    Traceback (most recent call last):
  File "/home/codespace/.local/lib/python3.12/site-packages/starlette/routing.py", line 693, in lifespan
    async with self.lifespan_context(app) as maybe_state:
  File "/usr/local/python/3.12.1/lib/python3.12/contextlib.py", line 210, in __aenter__
    return await anext(self.gen)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/codespace/.local/lib/python3.12/site-packages/fastapi/routing.py", line 133, in merged_lifespan
    async with original_context(app) as maybe_original_state:
  File "/usr/local/python/3.12.1/lib/python3.12/contextlib.py", line 210, in __aenter__
    return await anext(self.gen)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/codespace/.local/lib/python3.12/site-packages/fastapi/routing.py", line 133, in merged_lifespan
    async with original_context(app) as maybe_original_state:
  File "/usr/local/python/3.12.1/lib/python3.12/contextlib.py", line 210, in __aenter__
    return await anext(self.gen)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/codespace/.local/lib/python3.12/site-packages/fastapi/routing.py", line 133, in merged_lifespan
    async with original_context(app) as maybe_original_state:
  File "/usr/local/python/3.12.1/lib/python3.12/contextlib.py", line 210, in __aenter__
    return await anext(self.gen)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/codespace/.local/lib/python3.12/site-packages/fastapi/routing.py", line 134, in merged_lifespan
    async with nested_context(app) as maybe_nested_state:
  File "/usr/local/python/3.12.1/lib/python3.12/contextlib.py", line 210, in __aenter__
    return await anext(self.gen)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/python/3.12.1/lib/python3.12/contextlib.py", line 210, in __aenter__
    return await anext(self.gen)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/codespace/.local/lib/python3.12/site-packages/litserve/server.py", line 260, in lifespan
    response_queue = self.response_queues[app.response_queue_id]
                                          ^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'FastAPI' object has no attribute 'response_queue_id'
```

cc: @aniketmaurya

Reply:
checking now